server/mailet/mailets/src/main/java/org/apache/james/transport/util/SpecialAddressesUtils.java

```diff
@@ -186,6 +190,7 @@ public SpecialAddressesUtils(RedirectNotify genericMailet) {
             case SENDER:
             case REVERSE_PATH:
                 return Optional.ofNullable(mail.getSender())
+                    .filter(address -> !address.isNullSender())
```

Comment:
well, it looks like we repeat ourselves. What about `mail.getNonNullSenders()` ? why don't we have to check `replyTo.equals(SpecialAddress.NULL)` this time ?

Reply:
Again, I don't understand that code... Just did the changes where needed to get :green_apple:  tests